notebook/nbextensions.py

```diff
@@ -20,63 +23,30 @@
     from urlparse import urlparse
     from urllib import urlretrieve
 
-from jupyter_core.paths import jupyter_data_dir, jupyter_path, SYSTEM_JUPYTER_PATH
+from jupyter_core.paths import (
+    jupyter_data_dir, jupyter_path, jupyter_config_dir, jupyter_config_path,
+    SYSTEM_JUPYTER_PATH, ENV_JUPYTER_PATH, ENV_CONFIG_PATH, SYSTEM_CONFIG_PATH
+)
 from ipython_genutils.path import ensure_dir_exists
-from ipython_genutils.py3compat import string_types, cast_unicode_py2
+from ipython_genutils.py3compat import string_types, cast_unicode_py2, PY3
 from ipython_genutils.tempdir import TemporaryDirectory
 from ._version import __version__
 
-class ArgumentConflict(ValueError):
-    pass
+from traitlets.config.manager import BaseJSONConfigManager
 
+from tornado.log import LogFormatter
 
-def _should_copy(src, dest, verbose=1):
-    """should a file be copied?"""
-    if not os.path.exists(dest):
-        return True
-    if os.stat(src).st_mtime - os.stat(dest).st_mtime > 1e-6:
-        # we add a fudge factor to work around a bug in python 2.x
-        # that was fixed in python 3.x: http://bugs.python.org/issue12904
-        if verbose >= 2:
-            print("%s is out of date" % dest)
-        return True
-    if verbose >= 2:
-        print("%s is up to date" % dest)
-    return False
 
+#------------------------------------------------------------------------------
+# Public API
+#------------------------------------------------------------------------------
 
-def _maybe_copy(src, dest, verbose=1):
-    """copy a file if it needs updating"""
-    if _should_copy(src, dest, verbose):
-        if verbose >= 1:
-            print("copying %s -> %s" % (src, dest))
-        shutil.copy2(src, dest)
 
-def _safe_is_tarfile(path):
-    """safe version of is_tarfile, return False on IOError"""
-    try:
-        return tarfile.is_tarfile(path)
-    except IOError:
-        return False
-
-
-def _get_nbext_dir(nbextensions_dir=None, user=False, prefix=None):
```

Comment:
I know that `ipywidgets` has a dependency on `_get_nbext_dir` so we might want to keep this in mind as removing this would break the current version of `ipywidgets`.